crypto/src/main/scala/org/bitcoins/crypto/BouncyCastleUtil.scala

```diff
@@ -215,3 +220,224 @@ object BouncyCastleUtil {
     }
   }
 }
+
+object AdaptorStuff {
+  import ECAdaptorSignature.{deserializePoint, serializePoint}
+
+  // Compute s' = k^-1 * (dataToSign + rx*privateKey)
+  private def adaptorSignHelper(
+      dataToSign: ByteVector,
+      k: FieldElement,
+      r: ECPublicKey,
+      privateKey: ECPrivateKey): FieldElement = {
+    val rx = FieldElement(r.toPoint.getXCoord.toBigInteger)
+    val x = privateKey.fieldElement
+    val m = FieldElement(dataToSign)
+    val kInv = k.inverse
+
+    rx.multiply(x).add(m).multiply(kInv)
+  }
+
+  def adaptorSign(
+      privateKey: ECPrivateKey,
+      adaptorPoint: ECPublicKey,
+      dataToSign: ByteVector): ECAdaptorSignature = {
+    // Include dataToSign and adaptor in nonce derivation
+    val hash = CryptoUtil.sha256(dataToSign ++ serializePoint(adaptorPoint))
+    val k = DLEQStuff.dleqNonceFunc(hash.bytes,
+                                    privateKey.fieldElement,
+                                    "ECDSAAdaptorNon")
```

Comment:
Is this a special tag you added? If so should this be documented or at least factored out into a constant?

Reply:
Jonas made this

Reply:
(it'll be in Lloyd's spec)